packs/boundary_worker/README.md

```diff
@@ -0,0 +1,126 @@
+| app_enable_hcp_boundary_support          | Toggle to enable HCP Boundary Support (and forego self-hosted Boundary Enterprise Cluster registration workflows.  | `true` |
+| app_initial_upstreams                    | Initial Upstreams for the Boundary Worker.                                                                         | n/a |
+| app_tls_cert_file                        | Specifies the path to the certificate for the Boundary Worker.                                                     | n/a |
+| app_tls_cipher_suites                    | Overridden List of supported ciphersuites for the Boundary Worker.                                                 | n/a |
+| app_tls_client_ca_file                   | PEM-encoded Certificate Authority File used for checking the authenticity of tthe client for the Boundary Worker.  | n/a |
+| app_tls_disable                          | Toggle to disable TLS support for the Boundary Worker.                                                             | `true` |
+| app_tls_key_file                         | Specifies the path to the private key for the certificate for the Boundary Worker.                                 | n/a |
+| app_tls_max_version                      | Specifies the maximum supported TLS version for the Boundary Worker.                                               | `"tls13"` |
+| app_tls_min_version                      | Specifies the minimum supported TLS version for the Boundary Worker.                                               | `"tls12"` |
+| app_tls_prefer_server_cipher_suites      | Toggle to enable preference for Server's ciphersuites over Client's ciphersuites for the Boundary Worker.          | `false` |
+| app_tls_require_and_verify_client_cert   | Toggle to enable client authentication for the listener for the Boundary Worker.                                   | `false` |
+| app_worker_description                   | Description for the Boundary Worker.                                                                               | `"Nomad-managed Boundary Worker."` |
+| app_worker_name_prefix                   | Prefix for the Boundary Worker Name.                                                                               | `"nomad"` |
+
+### Nomad
+
+This section describes Nomad-specific configuration.
+
+| Name                              | Description                                  | Default |
+| --------------------------------- | -------------------------------------------- | ------- |
+| nomad_group_count                 | Count of Deployments for the Group.          | `1` |
+| nomad_group_ephemeral_disk        | Ephemeral Disk Configuration for the Group.  | `{"migrate":true,"size":128,"sticky":true}` |
+| nomad_group_name                  | Name for the Group.                          | `"boundary_worker"` |
+| nomad_group_network_mode          | Network Mode for the Group.                  | `"host"` |
```

Comment:
Might be worth mentioning that without TLS and with network_mode=host, anything on the host would be able to MITM/eavesdrop on the communications between the worker and the control plane, or making the defaults such as that it's hard (maybe an auto generated cert?)

Reply:
This is a good call out, I'll think about this more in #24